File: app/src/main/java/com/example/coursework_3course_budgetapp/OperationDao.java
package com.example.coursework_3course_budgetapp;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface OperationDao {

    @Query("SELECT * FROM operation")
    List<Operation> getAll();

    @Insert
    void insert(Operation operation);

    @Delete
    void delete(Operation operation);

    @Update
    void update(Operation operation);
}

File: app/src/main/java/com/example/coursework_3course_budgetapp/OperationDao.kt
package com.example.coursework_3course_budgetapp

import androidx.room.*

@Dao
interface OperationDao {
    @get:Query("SELECT * FROM Operations")
    val all: List<Operation?>?

    @Insert
    fun insert(operation: Operation?)

    @Delete
    fun delete(operation: Operation?)

    @Update
    fun update(operation: Operation?)
}